test/extensions/clusters/common/dns_cluster_backcompat_test.cc

```diff
@@ -0,0 +1,126 @@
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/extensions/clusters/common/dns/v3/dns.pb.h"
+#include "envoy/extensions/clusters/dns/v3/dns_cluster.pb.h"
+
+#include "source/extensions/clusters/common/dns_cluster_backcompat.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class DnsClusterBackcompatUtilTest : public testing::Test {};
+
+TEST_F(DnsClusterBackcompatUtilTest, Empty) {
+  envoy::config::cluster::v3::Cluster cluster{};
+  envoy::extensions::clusters::dns::v3::DnsCluster dns_cluster{};
+  createDnsClusterFromLegacyFields(cluster, dns_cluster);
+  ASSERT_FALSE(dns_cluster.has_dns_jitter());
```

Comment:
@Stevenjin8 nit, I don't know if it is a preference in Envoy code base specifically, but IMO, it's better to see all errors rather than getting and fixing them one by one, with that in mind, I would suggest to prefer EXPECT_ checks rather than ASSERT_ checks (i.e., EXPECT_ checks are non-fatal and will fail the test, but will allow it to finish first).